File: web/src/lib/validation.ts
export const MAX_ROOM_NAME_LENGTH = 80;
export const MAX_DISPLAY_NAME_LENGTH = 40;
export const MAX_PASSWORD_LENGTH = 256;
export const MAX_CHAT_MESSAGE_LENGTH = 4000;
export const MAX_FILE_SIZE = 25 * 1024 * 1024;

export function cleanBounded(value: string, max: number): string {
  return value.trim().slice(0, max);
}

export function fitsBound(value: string, max: number): boolean {
  return value.length <= max;
}

File: internal/hub/validation.go
package hub

import (
	"errors"
	"strings"
)

const (
	MaxRoomNameLength    = 80
	MaxDisplayNameLength = 40
	MaxPasswordLength    = 256
	MaxChannelIDLength   = 64
	MaxChannelNameLength = 80
	MaxSDPLength         = 32768
	MaxCandidateLength   = 8192
	MaxSignalNonceLength = 64
	MaxSignalSigLength   = 128
)

var errInvalidInput = errors.New("invalid input")

func cleanBounded(value string, max int) (string, bool) {
	value = strings.TrimSpace(value)
	if value == "" || len(value) > max {
		return "", false
	}
	return value, true
}

func validateOptionalBounded(value string, max int) bool {
	return len(value) <= max
}
